offer-manager-domain/src/main/groovy/life/qbic/business/offers/OfferContent.groovy

```diff
@@ -0,0 +1,319 @@
+            this.projectManagerTitle = Objects.requireNonNull(projectManagerTitle, "Projectmanager must not be null")
+            projectManagerEmail = Objects.requireNonNull(projectManager.emailAddress, "Projectmanager must not be null")
+
+            /*Projectinformation*/
+            this.creationDate = Objects.requireNonNull(creationDate, "Creation date must not be null")
+            this.expirationDate = Objects.requireNonNull(expirationDate, "Expiration date must not be null")
+            this.projectTitle = Objects.requireNonNull(projectTitle, "Project title must not be null")
+            this.projectObjective = Objects.requireNonNull(projectObjective, "Project objective must not be null")
+            this.experimentalDesign = Objects.requireNonNull(experimentalDesign, "Experimental design must not be  null")
+            this.offerIdentifier = Objects.requireNonNull(offerIdentifier, "Offer identifier must not be null")
+
+        }
+        Builder dataGenerationItems(List<OfferItem> dataGenerationItems){
+            this.dataGenerationItems = dataManagementItems
+            return this
+        }
+        Builder dataAnalysisItems(List<OfferItem> dataAnalysisItems){
+            this.dataAnalysisItems = dataAnalysisItems
+            return this
+        }
+        Builder dataManagementItems(List<OfferItem> dataManagementItems){
+            this.dataManagementItems = dataManagementItems
+            return this
+        }
+        Builder overheadTotal(double overheadTotal){
+            this.overheadTotal = overheadTotal
+            return this
+        }
+        Builder overheadsDataGeneration(double overheadDG){
+            this.overheadsDataGeneration = overheadDG
+            return this
+        }
+        Builder overheadsDataAnalysis(double overheadDA){
+            this.overheadsDataAnalysis = overheadDA
+            return this
+        }
+        Builder overheadsPMandDS(double overheadPMandDS){
+            this.overheadsPMandDS = overheadPMandDS
+            return this
+        }
+        Builder netDataGeneration(double net){
+            this.netDataGeneration = net
+            return this
+        }
+        Builder netDataAnalysis(double net){
+            this.netDataAnalysis = net
+            return this
+        }
+        Builder netProjectManagementAndDataStorage(double net){
+            this.netPMandDS = net
+            return this
+        }
+        Builder totalCost(double total){
+            this.totalCost = total
+            return this
+        }
+        Builder netCost(double net){
+            this.netCost = net
+            return this
+        }
+        Builder totalVat(double vat){
+            this.totalVat = vat
+            return this
+        }
+
+        OfferContent build(){
+            //require all fields to be set before the object can be created
+            if(dataGenerationItems == null) throw new IllegalArgumentException("Missing data generation items")
```

Comment:
I think you should throw NullPointerExceptions instead of IllegalArgumentExceptions since the build method does not take any arguments.

Reply:
Hm I see, was choosing between that one and the illegalargumentexception 🙈

Reply:
You could also use Object.requireNonNull with an appropriate message